### PR TITLE
Feat/add new timestamp to khata

### PR DESCRIPTION
**Pull request by chandreshkkhatri**

### CODE REVIEW

File: routes/transactions.js
Comment (shital01) [inline routes/transactions.js:1]: timestamp toh ek se hi kaam chal jayega seperate variable ki jarurat nahin padegi transaction ka created/updated timestamp hi khata wale table me jayega toh consistency issue nahin hoga

Comment (shital01) [inline routes/transactions.js:1]: var transactionsUpdatedAfterTimeStamp = req.query.transactionsUpdatedAfterTimeStamp ?? 0; 
variable name change ho gya last api se toh consistency issue aa jayega ,have a look at it

Comment (shital01) [inline routes/transactions.js:1]: also for get api in this particular case as we discussed there wont be need for v2 so you can remove that

Comment (shital01) [inline routes/transactions.js:1]: khata.lastTransactionUpdatedTimeStamp = transaction.updatedTimeStamp;
				await khata.save();

isme khata fetch karte time lastTransactionUpdateTimeStamp bhi fetch karlo and if new transaction .updateTimeStamp>khata.lastTransaction..... toh update and save karna h kam ke case me nahin basicaly latest timestamp rahegi isse

File: utils/validations/transactionValidations.js
Comment (shital01) [inline utils/validations/transactionValidations.js:1]: lastUpdatedTimeStamp: Joi.date().timestamp('unix'),
    transactionsUpdatedAfterTimeStamp: Joi.date().timestamp('unix'),

again first wale se hi kaam chal jayega

File: routes/transactions.js
Comment (chandreshkkhatri) [reply routes/transactions.js:1]: Since we are removing this query parameter, it should not matter anyways

Comment (chandreshkkhatri) [reply routes/transactions.js:1]: changed.

Comment (chandreshkkhatri) [reply routes/transactions.js:1]: add the condition for update

File: utils/validations/transactionValidations.js
Comment (chandreshkkhatri) [reply utils/validations/transactionValidations.js:1]: changed

File: routes/transactions.js [hunk 1]
```diff
@@ -54,20 +55,30 @@ router.get(
 				{ userPhoneNumber: { $eq: PhoneNumber } },
 				{ friendPhoneNumber: { $eq: PhoneNumber } },
 			],
-		}).select('_id');
+		}).lean();
+
```
Comment (chandreshkkhatri) [inline routes/transactions.js:59]: lean method returns JSON object instead of mongoose model

Comment (chandreshkkhatri) [inline routes/transactions.js:59]: lean method returns JSON object instead of mongoose model

Comment (chandreshkkhatri) [inline routes/transactions.js:59]: lean method returns JSON object instead of mongoose model

File: routes/transactions.js [hunk 2]
```diff
@@ -423,6 +451,13 @@ router.put(
 				);
 				if (!khata) {
 					continue;
+				} else {
+					if (khata.lastTransactionUpdatedTimeStamp < transaction.updatedTimeStamp) {
```
Comment (chandreshkkhatri) [inline routes/transactions.js:455]: added if condition on khata timestamp update here as well

File: routes/transactions.js [hunk 3]
```diff
@@ -145,12 +156,21 @@ router.get(
 				{ userPhoneNumber: { $eq: PhoneNumber } },
 				{ friendPhoneNumber: { $eq: PhoneNumber } },
 			],
-		}).select('_id');
+		}).lean();
+
+		const filteredKhatas = khatas.filter((khata) => {
+			if (khata.lastTransactionUpdatedTimeStamp) {
+				return khata.lastTransactionUpdatedTimeStamp > cursorTimeStamp
```
Comment (shital01) [inline routes/transactions.js:163]: check varible name for cursortimeStamp

File: routes/transactions.js [hunk 4]
```diff
@@ -253,10 +273,15 @@ router.post('/multiple', auth, device, async (req, res) => {
 			try {
 				const savedEntry = await transaction.save();
 				savedEntries.push(savedEntry);
-				//send notification
 				const khata = await Khata.findById(transaction.khataId).select(
 					'userPhoneNumber friendPhoneNumber friendName',
 				);
+				if (khata.lastTransactionUpdatedTimeStamp < transaction.updatedTimeStamp) {
```
Comment (shital01) [inline routes/transactions.js:279]: isme error aayega kyunki upar .select me lastTransactionUPdatedTimestamp nhin h

File: routes/transactions.js [hunk 5]
```diff
@@ -423,6 +451,13 @@ router.put(
 				);
 				if (!khata) {
 					continue;
+				} else {
+					if (khata.lastTransactionUpdatedTimeStamp < transaction.updatedTimeStamp) {
+						await Khata.updateOne(
+							{ _id: khata._id },
+							{ $set: { lastTransactionUpdatedTimeStamp: transaction.updatedTimeStamp } },
```
Comment (shital01) [inline routes/transactions.js:458]: condition for check is missing for greater than timestamp

Comment (chandreshkkhatri) [reply routes/transactions.js:458]: for less than timestamp, no update to khata is needed, right?

File: routes/transactions.js (hunk 4 above)
Comment (chandreshkkhatri) [reply routes/transactions.js:279]: have added timestamp field name to select method argument

**Pull request merged**